Samples/AudioCreation/cs/AudioCreation/Scenario7_RecordPlayback.xaml.cs

```diff
@@ -0,0 +1,570 @@
+        /// </summary>
+        private unsafe void HandleIncomingAudio()
+        {
+            AudioFrame frame = frameOutputNode.GetFrame();
+
+            using (AudioBuffer buffer = frame.LockBuffer(AudioBufferAccessMode.Read))
+            using (IMemoryBufferReference reference = buffer.CreateReference())
+            {
+                byte* dataInBytes;
+                uint capacityInBytes;
+
+                // Get the buffer from the AudioFrame
+                ((IMemoryBufferByteAccess)reference).GetBuffer(out dataInBytes, out capacityInBytes);
+
+                if (capacityInBytes == 0)
+                {
+                    // we don't count zero-byte frames... and why do they ever happen???
+                    return;
+                }
+
+                // Must be multiple of 8 (2 channels, 4 bytes/float)
+                Debug.Assert((capacityInBytes & 0x7) == 0);
+
+                lastCapacityEncountered = capacityInBytes;
+                maxCapacityEncountered = Math.Max(maxCapacityEncountered, capacityInBytes);
+
+                uint bufferStart = 0;
+                if (maxIndex == 0)
+                {
+                    // if maxCapacityEncountered is greater than the audio graph buffer size, 
+                    // then the audio graph decided to give us a big backload of buffer content
+                    // as its first callback.  Not sure why it does this, but we don't want it,
+                    // so take only the tail of the buffer.
+                    if (maxCapacityEncountered > (graph.LatencyInSamples << 3))
+                    {
+                        bufferStart = capacityInBytes - (uint)(graph.LatencyInSamples << 3);
+                        capacityInBytes = (uint)(graph.LatencyInSamples << 3);
+                    }
+                }
+
+                // if we fill up, just spin forever.  don't try to exit recording from audio thread,
+                // we don't bother with two-way signaling.
+                if (maxIndex + capacityInBytes > byteBuffer.Length)
+                {
+                    Debug.Assert(byteBuffer.Length >= maxIndex);
+                    capacityInBytes = (uint)byteBuffer.Length - maxIndex;
+                }
+
+                fixed (byte* buf = byteBuffer)
+                {
+                    for (int i = 0; i < capacityInBytes; i++)
+                    {
+                        byte b = dataInBytes[i + bufferStart];
+                        buf[i + maxIndex] = b;
+
+                        if (b != 0)
+                        {
+                            nonZeroSampleCount++;
+                        }
+                    }
+                }
+                maxIndex += capacityInBytes;
+            }
+
+            audioFrameCount++;
+        }
+
+        private void UpdateStatusWhileRecording()
+        {
+            uint max = maxIndex >> 3;
+            uint curr = currentIndex >> 3;
```

Comment:
I don't see you using the variable "curr", so it can be removed. Also the shift by 3 is converting from bytes to number of samples, but perhaps that is not clear to everyone so a comment would be useful.